src/Compilers/CSharp/Test/Emit/CodeGen/IndexAndRangeTests.cs

```diff
@@ -1,12 +1,685 @@
 ﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
 
 using Microsoft.CodeAnalysis.CSharp.Test.Utilities;
+using Microsoft.CodeAnalysis.Test.Utilities;
 using Xunit;
 
 namespace Microsoft.CodeAnalysis.CSharp.UnitTests.CodeGen
 {
     public class IndexAndRangeTests : CSharpTestBase
     {
+        [Fact]
+        public void IndexIndexerStringTwoArgs()
+        {
+            var comp = CreateCompilationWithIndex(@"
+using System;
+class C
+{
+    public static void Main()
+    {
+        var s = ""abcdef"";
+        M(s);
+    }
+    public static void M(string s)
+    {
+        Console.WriteLine(s[new Index(1, false)]);
+        Console.WriteLine(s[new Index(1, false), ^1]);
+    }
+}");
+            comp.VerifyDiagnostics(
+                // (13,27): error CS1501: No overload for method 'this' takes 2 arguments
+                //         Console.WriteLine(s[new Index(1, false), ^1]);
+                Diagnostic(ErrorCode.ERR_BadArgCount, "s[new Index(1, false), ^1]").WithArguments("this", "2").WithLocation(13, 27));
+        }
+
+        [Fact]
+        public void IndexIndexerArrayTwoArgs()
+        {
+            var comp = CreateCompilationWithIndex(@"
+using System;
+class C
+{
+    public static void Main()
+    {
+        var x = new int[1,1];
+        M(x);
+    }
+    public static void M(int[,] s)
+    {
+        Console.WriteLine(s[new Index(1, false), ^1]);
+    }
+}");
+            comp.VerifyDiagnostics(
+                // (12,27): error CS0029: Cannot implicitly convert type 'System.Index' to 'int'
+                //         Console.WriteLine(s[new Index(1, false), ^1]);
+                Diagnostic(ErrorCode.ERR_NoImplicitConv, "s[new Index(1, false), ^1]").WithArguments("System.Index", "int").WithLocation(12, 27),
+                // (12,27): error CS0029: Cannot implicitly convert type 'System.Index' to 'int'
+                //         Console.WriteLine(s[new Index(1, false), ^1]);
+                Diagnostic(ErrorCode.ERR_NoImplicitConv, "s[new Index(1, false), ^1]").WithArguments("System.Index", "int").WithLocation(12, 27));
+        }
+
+        [Fact]
+        public void FakeIndexIndexerString()
```

Comment:
>Fake [](start = 20, length = 4)

The prefix seems unnecessary on these tests. Won't we keep the same tests even when the work around is removed?

Reply:
Not necessarily. We don't normally include framework tests in our source.